src/RequestHandler/inprocess/inprocessapplication.cpp

```diff
@@ -507,6 +507,45 @@ IN_PROCESS_APPLICATION::ExecuteAspNetCoreProcess(
 
 }
 
+HRESULT
+IN_PROCESS_APPLICATION::SetEnvironementVariablesOnWorkerProcess(
+    VOID
+)
+{
+    HRESULT hr = S_OK;
+    ENVIRONMENT_VAR_HASH* pEnvironmentVarTable;
+    ENVIRONMENT_VAR_ENTRY* pIISAuthEntry = NULL;
+
+    pEnvironmentVarTable = new ENVIRONMENT_VAR_HASH();
+    if (pEnvironmentVarTable == NULL)
+    {
+        hr = E_OUTOFMEMORY;
+        goto Finished;
+    }
+
+    //
+    // few environment variables expected, small bucket size for hash table
+    //
+    if (FAILED(hr = pEnvironmentVarTable->Initialize(37 /*prime*/)))
+    {
+        goto Finished;
+    }
+
+    m_pConfig->QueryEnvironmentVariables()->Apply(ENVIRONMENT_VAR_HASH::CopyToTable, pEnvironmentVarTable);
+    pEnvironmentVarTable->FindKey(ASPNETCORE_IIS_AUTH_ENV_STR, &pIISAuthEntry);
```

Comment:
how about others such as ASPNETCORE_PORT, ASPNETCORE_APPL_PATH? Does customer code use them?

Reply:
I modified the logic to now include the ASPNETCORE_AUTH_ENV string.